install.sh

```diff
@@ -605,220 +597,192 @@ prepare_environment_for_buildah() {
-        run_as_root "apt-get install -y uidmap"
-    elif command -v yum &> /dev/null; then
-        run_as_root "yum install -y shadow-utils"
-    elif command -v zypper &> /dev/null; then
-        run_as_root "zypper install -y shadow"
-    else
-        log::info "Please install newuidmap and newgidmap manually for your distribution."
-        exit 1
-    fi
+install_buildah(){
+  distro=$(get_linux_distro)
+  log::info "Installing Buildah and configuring user namespaces..."
 
-    # Ensure correct capabilities for newuidmap and newgidmap
-    log::info "Setting capabilities for newuidmap and newgidmap..."
-    run_as_root "setcap cap_setuid+ep /usr/bin/newuidmap"
-    run_as_root "setcap cap_setgid+ep /usr/bin/newgidmap"
-    run_as_root "chmod u-s,g-s /usr/bin/newuidmap /usr/bin/newgidmap"
-
-    # Ensure /etc/subuid and /etc/subgid files exist
-    log::info "Checking /etc/subuid and /etc/subgid..."
-    if [ ! -f /etc/subuid ] || [ ! -f /etc/subgid ]; then
-        if command -v apt-get &> /dev/null; then
-            run_as_root "apt-get install -y login"
-        elif command -v yum &> /dev/null ; then
-            run_as_root "yum install -y shadow-utils"
-        elif command -v zypper &> /dev/null; then
-            run_as_root "zypper install -y shadow"
-        else
-            echo "Please ensure /etc/subuid and /etc/subgid are correctly set up."
-            exit 1
-        fi
-    fi
-
-    # Add an entry to /etc/subuid and /etc/subgid
-    log::info "Configuring /etc/subuid and /etc/subgid for user $CURRENT_USER..."
-    if ! grep -q "$CURRENT_USER" /etc/subuid; then
-        echo "$CURRENT_USER:1000000:65536" | run_as_root "tee -a /etc/subuid"
-    fi
-    if ! grep -q "$CURRENT_USER" /etc/subgid; then
-        echo "$CURRENT_USER:1000000:65536" | run_as_root "tee -a /etc/subgid"
-    fi
+  case "$distro" in
+    *Debian*|*Ubuntu*)
+      install_package "buildah uidmap"
+      ;;
+    *CentOS*|*Red\ Hat*|*Fedora*)
+      run_as_root "yum groupinstall -y 'Development Tools'"
+      install_package "curl-devel expat-devel gettext-devel openssl-devel zlib-devel perl autoconf"
+      ;;
+    *SUSE*|*openSUSE*)
+      install_package "buildah shadow"
+      ;;
+    *RHEL*)
+      if [[ "$distro" == *"8"* ]]; then
+          run_as_root "dnf module enable -y container-tools:2.0"
+          install_package "buildah uidmap"
+      elif [[ "$distro" == *"7"* ]]; then
+          run_as_root "subscription-manager repos --enable=rhel-7-server-extras-rpms"
+          install_package "buildah uidmap"
+      fi
+      ;;
+    *)
+      abort "Unsupported distribution."
+      ;;
+  esac
 
-    log::info "Manual configuration complete. You may need to reboot your system for changes to take full effect."
-}
+  # Configure user namespaces
+  if ! grep -q "$(get_user)" /etc/subuid; then
+    log::info "Configuring /etc/subuid and /etc/subgid for user namespaces..."
+    echo "$(get_user):100000:65536" | sudo tee -a /etc/subuid /etc/subgid
```

Comment:
There might be collision with already defined entries in these files. There should be some other way to add these entries, which respects already allocated ranges.